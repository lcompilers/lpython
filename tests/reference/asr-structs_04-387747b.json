{
    "basename": "asr-structs_04-387747b",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/structs_04.py",
    "infile_hash": "87ec2ee343d4931ce152501fa1d36db33279b93a3d1b25bc4463b0da",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-structs_04-387747b.stdout",
    "stdout_hash": "4c781c5f2d3fe44166b39d672854919409d7390fcd4d39e3afe7700e",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}