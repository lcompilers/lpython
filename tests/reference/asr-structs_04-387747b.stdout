(TranslationUnit
    (SymbolTable
        1
        {
            __main__:
                (Module
                    (SymbolTable
                        2
                        {
                            A:
                                (StructType
                                    (SymbolTable
                                        3
                                        {
                                            x:
                                                (Variable
                                                    3
                                                    x
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            y:
                                                (Variable
                                                    3
                                                    y
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    A
                                    []
                                    [y
                                    x]
                                    Source
                                    Public
                                    .false.
                                    .false.
                                    [(())
                                    (())]
                                    ()
                                    ()
                                ),
                            B:
                                (StructType
                                    (SymbolTable
                                        4
                                        {
                                            a:
                                                (Variable
                                                    4
                                                    a
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Struct
                                                        2 A
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            z:
                                                (Variable
                                                    4
                                                    z
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    B
                                    [A]
                                    [z
                                    a]
                                    Source
                                    Public
                                    .false.
                                    .false.
                                    [(())
                                    ((StructTypeConstructor
                                        2 A
                                        [((Cast
                                            (RealConstant
                                                0.000000
                                                (Real 8)
                                            )
                                            RealToReal
                                            (Real 4)
                                            (RealConstant
                                                0.000000
                                                (Real 4)
                                            )
                                        ))
                                        ((IntegerConstant 0 (Integer 4)))]
                                        (Struct
                                            2 A
                                        )
                                        ()
                                    ))]
                                    ()
                                    ()
                                ),
                            __main__global_stmts:
                                (Function
                                    (SymbolTable
                                        9
                                        {
                                            
                                        })
                                    __main__global_stmts
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [g]
                                    []
                                    [(SubroutineCall
                                        2 g
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            f:
                                (Function
                                    (SymbolTable
                                        5
                                        {
                                            b:
                                                (Variable
                                                    5
                                                    b
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (Struct
                                                        2 B
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    f
                                    (FunctionType
                                        [(Struct
                                            2 B
                                        )]
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 5 b)]
                                    [(Print
                                        [(StructInstanceMember
                                            (Var 5 b)
                                            4 z
                                            (Integer 4)
                                            ()
                                        )
                                        (StructInstanceMember
                                            (StructInstanceMember
                                                (Var 5 b)
                                                4 a
                                                (Struct
                                                    2 A
                                                )
                                                ()
                                            )
                                            3 x
                                            (Integer 4)
                                            ()
                                        )
                                        (StructInstanceMember
                                            (StructInstanceMember
                                                (Var 5 b)
                                                4 a
                                                (Struct
                                                    2 A
                                                )
                                                ()
                                            )
                                            3 y
                                            (Real 4)
                                            ()
                                        )]
                                        ()
                                        ()
                                    )
                                    (Assert
                                        (IntegerCompare
                                            (StructInstanceMember
                                                (Var 5 b)
                                                4 z
                                                (Integer 4)
                                                ()
                                            )
                                            Eq
                                            (IntegerConstant 1 (Integer 4))
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assert
                                        (IntegerCompare
                                            (StructInstanceMember
                                                (StructInstanceMember
                                                    (Var 5 b)
                                                    4 a
                                                    (Struct
                                                        2 A
                                                    )
                                                    ()
                                                )
                                                3 x
                                                (Integer 4)
                                                ()
                                            )
                                            Eq
                                            (IntegerConstant 2 (Integer 4))
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assert
                                        (RealCompare
                                            (Cast
                                                (StructInstanceMember
                                                    (StructInstanceMember
                                                        (Var 5 b)
                                                        4 a
                                                        (Struct
                                                            2 A
                                                        )
                                                        ()
                                                    )
                                                    3 y
                                                    (Real 4)
                                                    ()
                                                )
                                                RealToReal
                                                (Real 8)
                                                ()
                                            )
                                            Eq
                                            (RealConstant
                                                3.000000
                                                (Real 8)
                                            )
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            g:
                                (Function
                                    (SymbolTable
                                        6
                                        {
                                            a1:
                                                (Variable
                                                    6
                                                    a1
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Struct
                                                        2 A
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            a2:
                                                (Variable
                                                    6
                                                    a2
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Struct
                                                        2 A
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            b:
                                                (Variable
                                                    6
                                                    b
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Struct
                                                        2 B
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    g
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [f]
                                    []
                                    [(Assignment
                                        (Var 6 a1)
                                        (StructTypeConstructor
                                            2 A
                                            [((Cast
                                                (RealConstant
                                                    1.000000
                                                    (Real 8)
                                                )
                                                RealToReal
                                                (Real 4)
                                                (RealConstant
                                                    1.000000
                                                    (Real 4)
                                                )
                                            ))
                                            ((IntegerConstant 1 (Integer 4)))]
                                            (Struct
                                                2 A
                                            )
                                            ()
                                        )
                                        ()
                                    )
                                    (Assignment
                                        (Var 6 a2)
                                        (StructTypeConstructor
                                            2 A
                                            [((Cast
                                                (RealConstant
                                                    2.000000
                                                    (Real 8)
                                                )
                                                RealToReal
                                                (Real 4)
                                                (RealConstant
                                                    2.000000
                                                    (Real 4)
                                                )
                                            ))
                                            ((IntegerConstant 2 (Integer 4)))]
                                            (Struct
                                                2 A
                                            )
                                            ()
                                        )
                                        ()
                                    )
                                    (Assignment
                                        (Var 6 b)
                                        (StructTypeConstructor
                                            2 B
                                            [((IntegerConstant 1 (Integer 4)))
                                            ((Var 6 a1))]
                                            (Struct
                                                2 B
                                            )
                                            ()
                                        )
                                        ()
                                    )
                                    (Assignment
                                        (StructInstanceMember
                                            (Var 6 b)
                                            4 a
                                            (Struct
                                                2 A
                                            )
                                            ()
                                        )
                                        (Var 6 a2)
                                        ()
                                    )
                                    (Assignment
                                        (StructInstanceMember
                                            (Var 6 b)
                                            4 z
                                            (Integer 4)
                                            ()
                                        )
                                        (IntegerConstant 1 (Integer 4))
                                        ()
                                    )
                                    (Assignment
                                        (StructInstanceMember
                                            (StructInstanceMember
                                                (Var 6 b)
                                                4 a
                                                (Struct
                                                    2 A
                                                )
                                                ()
                                            )
                                            3 x
                                            (Integer 4)
                                            ()
                                        )
                                        (IntegerConstant 2 (Integer 4))
                                        ()
                                    )
                                    (Assignment
                                        (StructInstanceMember
                                            (StructInstanceMember
                                                (Var 6 b)
                                                4 a
                                                (Struct
                                                    2 A
                                                )
                                                ()
                                            )
                                            3 y
                                            (Real 4)
                                            ()
                                        )
                                        (Cast
                                            (RealConstant
                                                3.000000
                                                (Real 8)
                                            )
                                            RealToReal
                                            (Real 4)
                                            (RealConstant
                                                3.000000
                                                (Real 4)
                                            )
                                        )
                                        ()
                                    )
                                    (Assert
                                        (IntegerCompare
                                            (StructInstanceMember
                                                (Var 6 a1)
                                                3 x
                                                (Integer 4)
                                                ()
                                            )
                                            Eq
                                            (IntegerConstant 1 (Integer 4))
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assert
                                        (RealCompare
                                            (Cast
                                                (StructInstanceMember
                                                    (Var 6 a1)
                                                    3 y
                                                    (Real 4)
                                                    ()
                                                )
                                                RealToReal
                                                (Real 8)
                                                ()
                                            )
                                            Eq
                                            (RealConstant
                                                1.000000
                                                (Real 8)
                                            )
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assert
                                        (IntegerCompare
                                            (StructInstanceMember
                                                (Var 6 a2)
                                                3 x
                                                (Integer 4)
                                                ()
                                            )
                                            Eq
                                            (IntegerConstant 2 (Integer 4))
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assert
                                        (RealCompare
                                            (Cast
                                                (StructInstanceMember
                                                    (Var 6 a2)
                                                    3 y
                                                    (Real 4)
                                                    ()
                                                )
                                                RealToReal
                                                (Real 8)
                                                ()
                                            )
                                            Eq
                                            (RealConstant
                                                2.000000
                                                (Real 8)
                                            )
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (SubroutineCall
                                        2 f
                                        ()
                                        [((Var 6 b))]
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    __main__
                    []
                    .false.
                    .false.
                ),
            main_program:
                (Program
                    (SymbolTable
                        10
                        {
                            __main__global_stmts:
                                (ExternalSymbol
                                    10
                                    __main__global_stmts
                                    2 __main__global_stmts
                                    __main__
                                    []
                                    __main__global_stmts
                                    Public
                                )
                        })
                    main_program
                    [__main__]
                    [(SubroutineCall
                        10 __main__global_stmts
                        2 __main__global_stmts
                        []
                        ()
                    )]
                )
        })
    []
)
